src/ameba/rule/style/verbose_block.cr

```diff
@@ -0,0 +1,162 @@
+  #
+  # ```
+  # (1..3).any? { |i| i.odd? }
+  # ```
+  #
+  # And it should be written as this:
+  #
+  # ```
+  # (1..3).any?(&.odd?)
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Style/VerboseBlock:
+  #   Enabled: true
+  #   ExcludeMultipleLineBlocks: true
+  #   ExcludeCallsWithBlocks: false
+  #   ExcludeOperators: false
+  #   ExcludeSetters: false
+  #   MaxLineLength: ~
+  #   MaxLength: 50 # use ~ to disable
+  # ```
+  class VerboseBlock < Base
+    properties do
+      description "Identifies usage of collapsible single expression blocks."
+
+      exclude_calls_with_block true
+      exclude_multiple_line_blocks false
+      exclude_operators false
+      exclude_setters false
+
+      max_line_length : Int32? = nil # 100
+      max_length : Int32? = 50
+    end
+
+    MSG          = "Use short block notation instead: `%s`"
+    CALL_PATTERN = "%s(%s&.%s)"
+
+    protected def same_location_lines?(a, b)
+      return unless a_location = a.name_location
+      return unless b_location = b.location
+
+      a_location.line_number == b_location.line_number
+    end
+
+    private OPERATOR_CHARS =
+      {'[', ']', '!', '=', '>', '<', '~', '+', '-', '*', '/', '%', '^', '|', '&'}
+
+    protected def operator?(name)
+      name.each_char do |char|
+        return false unless char.in?(OPERATOR_CHARS)
+      end
+      !name.empty?
+    end
+
+    protected def setter?(name)
+      !name.empty? && name[0].letter? && name.ends_with?('=')
+    end
+
+    protected def valid_length?(code)
+      if max_length = self.max_length
+        return code.size <= max_length
+      end
+      true
+    end
+
+    protected def valid_line_length?(node, code)
+      if max_line_length = self.max_line_length
+        if location = node.name_location
+          final_line_length = location.column_number + code.size
+          return final_line_length <= max_line_length
+        end
+      end
+      true
+    end
+
+    protected def node_to_s(node : Crystal::Call)
+      case name = node.name
+      when "[]"
+        name = "[#{node.args.join ", "}]"
+      when "[]?"
+        name = "[#{node.args.join ", "}]?"
+      when "[]="
+        unless node.args.empty?
+          name = "[#{node.args[..-2].join ", "}]=(#{node.args.last})"
+        end
+      else
+        name += "(#{node.args.join ", "})" unless node.args.empty?
+        name += " {...}" if node.block
+      end
+      name
+    end
+
+    protected def call_code(call, body)
+      args = call.args.join ", " unless call.args.empty?
+      args += ", " if args
+
+      call_chain = %w[].tap do |arr|
+        obj = body.obj
+        while obj.is_a?(Crystal::Call)
+          arr << node_to_s(obj)
+          obj = obj.obj
+        end
+        arr.reverse!
+        arr << node_to_s(body)
+      end
+
+      name =
+        call_chain.join('.')
+
+      CALL_PATTERN % {call.name, args, name}
+    end
+
+    # ameba:disable Metrics/CyclomaticComplexity
```

Comment:
I suggest splitting this into few self-explained smaller methods. Even without `Metrics/CyclomaticComplexity` check this method looks a bit complicated and messy.

Reply:
I'm open to a suggestions since I couldn't come up with anything less complicated for one method and easily readable/comprehensible at the same time.

Reply:
With the addition of `exclude_prefix_operators` option, issue is back again, and this time I really have no other idea how to avoid it.